src/test/ui/generator/borrowing.nll.stderr

```diff
@@ -10,15 +10,16 @@ LL |     };
 error[E0597]: `a` does not live long enough
   --> $DIR/borrowing.rs:24:9
    |
+LL |       let _b = {
+   |           -- borrow later used here
```

Comment:
So this message is generated here:

https://github.com/rust-lang/rust/blob/dfabe4b88509bfff1eca9060ccc9a4d28ee394e8/src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs#L79

Given that we have the location in the MIR in the `location` variable, it would be quite trivial for us to determine if this is a `ReadForMatch` statement. The only catch is that we use this same statement *both* for matches and for this. If we refactor `ReadForMatch` to be something like this:

```rust
FakeRead(FakeReadCause, Place<'tcx>),
```

with

```rust
enum FakeReadCause {
    ForMatch,
    ForLet,
}
```

then we could match on this particular case and give a different message. Perhaps something like

```
-- borrowed value is then later stored into this variable
```